src/garage/experiment/local_runner.py

```diff
@@ -165,39 +178,47 @@ def restore(self, from_dir, from_epoch='last'):
                 Not applicable when snapshot_mode='last'.
 
         Returns:
-            A SimpleNamespace for train()'s arguments.
+            types.SimpleNamespace: A SimpleNamespace for train()'s arguments.
 
         """
         saved = self._snapshotter.load(from_dir, from_epoch)
 
         self._setup_args = saved['setup_args']
-        self.train_args = saved['train_args']
+        self._train_args = saved['train_args']
+        self._stats = saved['stats']
+
+        set_seed(self._setup_args.seed)
 
         self.setup(env=saved['env'],
                    algo=saved['algo'],
                    sampler_cls=self._setup_args.sampler_cls,
                    sampler_args=self._setup_args.sampler_args)
 
-        n_epochs = self.train_args.n_epochs
-        last_epoch = saved['last_epoch']
-        n_epoch_cycles = self.train_args.n_epoch_cycles
-        batch_size = self.train_args.batch_size
-        store_paths = self.train_args.store_paths
-        pause_for_plot = self.train_args.pause_for_plot
+        n_epochs = self._train_args.n_epochs
+        last_epoch = self._stats.total_epoch
+        last_itr = self._stats.total_itr
+        total_env_steps = self._stats.total_env_steps
+        n_epoch_cycles = self._train_args.n_epoch_cycles
+        batch_size = self._train_args.batch_size
+        store_paths = self._train_args.store_paths
+        pause_for_plot = self._train_args.pause_for_plot
 
         fmt = '{:<20} {:<15}'
         logger.log('Restore from snapshot saved in %s' %
                    self._snapshotter.snapshot_dir)
-        logger.log(fmt.format('Train Args', 'Value'))
+        logger.log(fmt.format('-- Train Args --', '-- Value --'))
```

Comment:
hmm maybe we can add a `dowel` feature which recognizes a format string and implements this API...

Reply:
wait, in general you should use `tabular` for this kind of formatting. is there a reason you didn't?

Reply:
tabular will log these data to all the sinks (console, csv and tensorboard), but I only want to log them once at the start of the experiment to console. Otherwise, they will spoil the table fields.

Reply:
okay, i see. i didn't realize this was under `restore`

we should probably provide a way in dowel to only print a table once, but perhaps that's for another day.

do you log `total_env_steps` to tabular anywhere?

Reply:
It's logged in algorithms, before call to `train_once()`.

Reply:
why not log it here? are you trying to avoid having Runner log to tabular?

Reply:
Make sense. My original idea to follow the pattern that most logs are done in algorithms. I will try to do so. If I can't make it then let's leave it to future.

Reply:
I am blocked by off-policy algorithms when trying to move this log to the runner. 

Basically, an algorithm is in control of the log flow. For off-policy algorithms, they usually have some warm-up epochs, when they don't do any logging. This information is invisible to the runner. 

If `TotalEnvSteps` is logged in the runner, it will be the only thing logged at the first several epochs. And that will confuses both `CsvOuput` and `TensorboardOutput`.

Reply:
So I will just keep it logged in algorithms.

Reply:
okay thanks for trying. I think the right place to do this is in the sampler, to keep all algorithms honest.

Can you file an issue for that and assign it to KR?

Reply:
Will do.